### DISCUSSION TITLE
Simplest way to serve from source w/o rebuild?

### DISCUSSION THREAD

**Discussion by EricCousineau-TRI** [foxglove/community#28] [Q&A]
Is there a "best way" to serve in production mode when building from source, but only doing incremental / on-purpose builds?
From README, I see the usage of `yarn run web:serve`, but I see that seems to rebuild every time.

From looking at the `/Dockerfile`, I can do the following (as of 1ddd5b04):
```sh
cd studio
yarn install --immutable
yarn run web:build:prod
python -m http.server -d web/.webpack 8080
```

Is there a simpler way to do this w/ `web:serve` that I'm missing?
Just a curiosity question atm to get a feel for things and know my options.

---

As an aside, can also serve using `caddy`:
```sh
docker run --rm -p 8080:8080 -v ${PWD}/web/.webpack:/mnt -w /mnt caddy:2 caddy file-server --listen :8080
```

**Comment by jtbandes** [comment]
The [`web:serve` script](https://github.com/foxglove/studio/blob/2776fdef2e1b5cb35b1506d2c349b37cfcd3a840/package.json#L29) has a hard-coded `--mode development`, but that could be changed to `production` to serve in production mode. In fact, you might be able to just run `yarn web:serve --mode production` to override the previous flag, but I'm not sure if that will work.

As far as I know, `webpack serve` does use incremental builds by default. Have you measured the time for a cold build vs the time it takes to settle/update after modifying one file? While the incremental rebuild time may be slow, I think it is faster than a cold build (It may also depend on which file is modified.)

Out of curiosity, why are you trying to use serve with production mode?
